pkg/queue/tenant_queues.go

```diff
@@ -151,6 +154,20 @@ func (q *tenantQueues) getOrAddQueue(tenant string, path []string, maxQueriers i
 	return uq.add(path)
 }
 
+func (uq *tenantQueue) computeMaxQueriers(consumers int) int {
+	if uq.maxQueryCapacity == 0 {
+		return uq.maxQueriers
+	}
+
+	maxQueriers := int(math.Ceil(float64(consumers) * uq.maxQueryCapacity))
+
+	if uq.maxQueriers != 0 && uq.maxQueriers < maxQueriers {
+		return uq.maxQueriers
+	}
+
+	return maxQueriers
+}
```

Comment:
`tenantQueue` should not have an additional responsibility of calculating the max capacity, and know how to juggle between the two settings.

This code can also expect some subtle bugs in the future because it relies on something outside to set `uq.maxQueriers` = `max_queriers_per_tenant` correctly. Right now it works just fine, but after a refactor in future...?

To improve the safety of this code, I'd suggest creating a type which accepts all the required inputs (`max_queriers_per_tenant`, `max_query_capacity`) and returns the capacity with a function which accepts `consumers` as input. You can then externalise all this business logic, it'll be self-documenting, and still equally testable.